src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -442,18 +442,13 @@ internal static bool CanEmitConstant(object value, Type type)
             }
 
             Type t = value as Type;
-            if (t != null && ShouldLdtoken(t))
+            if (t != null)
```

Comment:
`CanEmitConstant` can't be removed as it's still used in a visitor, but can be slightly cleaned up here, and the comment about it matching `TryEmitConstant` is no longer untrue!